challenge-react/src/App.js

```diff
@@ -1,111 +1,221 @@
 import React, { Component } from 'react';
 import fetch from 'isomorphic-fetch';
-import styled from 'styled-components';
 import { connect } from 'react-redux';
 import { summaryDonations } from './helpers';
+import { styleCss } from './styling'
+import Alert from '@material-ui/lab/Alert';
 
-const Card = styled.div`
-  margin: 10px;
-  border: 1px solid #ccc;
-`;
 
-export default connect((state) => state)(
-  class App extends Component {
-    state = {
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
       charities: [],
       selectedAmount: 10,
     };
+    this.handlePay = this.handlePay.bind(this);
+    this.showHide = this.showHide.bind(this)
+    this.closePaymentModal = this.closePaymentModal.bind(this);
+  }
+
+  closePaymentModal() {
+    const { charities } = this.state;
+    charities.map((charity, i) => {
+      if (charity.show) {
+        charity.show = false;
+      }
+    });
+    this.setState( {charities})
+    // reset selected amount once click donate to another charity
+    this.setState( {selectedAmount: 10})
+  }
+
+
+  showHide(e) {
+    const { charities } = this.state;
+    const currentCharity= e.target.id -1;
+    // To make sure other payment modal is closed before open the current one
+    this.closePaymentModal();
+    charities[currentCharity].show = !charities[currentCharity].show ? true : false;
```

Comment:
- Using an impure operator.
- This code can be simpify like this `charities[currentCharity].show = !charities[currentCharity].show`.

Reply:
Good point